data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3Sink.java

```diff
@@ -0,0 +1,143 @@
+	private S3SinkWorker worker;
+	private SinkAccumulator accumulator;
+	private final Codec codec;
+	private volatile boolean initialized;
+	private static BlockingQueue<Event> eventQueue;
+	private static boolean isStopRequested;
+	private Thread workerThread;
+	
+	
+
+	/**
+	 * 
+	 * @param pluginSetting
+	 * @param s3SinkConfig
+	 * @param pluginFactory
+	 */
+	@DataPrepperPluginConstructor
+	public S3Sink(PluginSetting pluginSetting, final S3SinkConfig s3SinkConfig, final PluginFactory pluginFactory) {
+		super(pluginSetting);
+		this.s3SinkConfig = s3SinkConfig;
+		final PluginModel codecConfiguration = s3SinkConfig.getCodec();
+		final PluginSetting codecPluginSettings = new PluginSetting(codecConfiguration.getPluginName(), codecConfiguration.getPluginSettings());
+		codec = pluginFactory.loadPlugin(Codec.class, codecPluginSettings);
+		initialized = Boolean.FALSE;
+	}
+
+	@Override
+	public boolean isReady() {
+		return initialized;
+	}
+
+	@Override
+	public void doInitialize() {
+		try {
+			doInitializeInternal();
+		} catch (InvalidPluginConfigurationException e) {
+			LOG.error("Failed to initialize S3-Sink.");
+			this.shutdown();
+			throw new RuntimeException(e.getMessage(), e);
+		} catch (Exception e) {
+			LOG.warn("Failed to initialize S3-Sink, retrying. Error - {} \n {}", e.getMessage(), e.getCause());
+		}
+	}
+
+	private void doInitializeInternal() {
+		eventQueue = new ArrayBlockingQueue<>(EVENT_QUEUE_SIZE);
+		S3SinkService s3SinkService = new S3SinkService(s3SinkConfig);
+		worker = new S3SinkWorker(s3SinkService, s3SinkConfig, codec);
+		S3SinkWorkerRunner runner = new S3SinkWorkerRunner();
+		workerThread = new Thread(runner);
+		workerThread.start();
+		initialized = Boolean.TRUE;
+	}
+
+	@Override
+	public void doOutput(final Collection<Record<Event>> records) {
+		LOG.debug("Records size : {}", records.size());
+		if (records.isEmpty()) {
+			return;
+		}
+		
+		for (final Record<Event> recordData : records) {
+			Event event = recordData.getData();
+			getEventQueue().add(event);
+		}
+	}
+
+	@Override
+	public void shutdown() {
+		super.shutdown();
+		isStopRequested = Boolean.TRUE;
+		if (workerThread.isAlive()) {
+			workerThread.stop();
+		}
+		LOG.info("s3-sink sutdonwn completed");
+	}
+
+	public static BlockingQueue<Event> getEventQueue() {
+		return eventQueue;
+	}
+
+	public static boolean isStopRequested() {
+		return isStopRequested;
+	}
+	
+	private class S3SinkWorkerRunner implements Runnable {
+		@Override
+		public void run() {
+			try {
+				while (!S3Sink.isStopRequested()) {
+					if (s3SinkConfig.getTemporaryStorage().equalsIgnoreCase(IN_MEMORY)) {
+						accumulator = worker.inMemmoryAccumulator();
+					} else {
+						accumulator = worker.localFileAccumulator();
+					}
+					accumulator.doAccumulate();
+				}
+			} catch (Exception e) {
+				e.printStackTrace();
+				LOG.error("Exception in S3Sink : \n Error message {} \n Exception cause {}", e.getMessage(),

```

Comment:
I wonder if we should have newlines in the logs. I'm not sure if it is very common, except in the stack trace. I'm not against it, but more wondering if it is a good pattern here.

Reply:
Thanks, David for your review. Removed newlines from logs.